### PR TITLE
Sharethrough: Split multiformat bid request into 1 bid request per media type

### PR DESCRIPTION
**Pull request by maxime-dupuis**
Mirroring this [pull request](https://github.com/prebid/prebid-server/pull/2866) in the Golang adapter
Adresses: https://github.com/prebid/prebid-server-java/issues/2481

### CODE REVIEW

**Comment by maxime-dupuis** [comment]
The functional tests fail for this PR but also for other PR's. So I think this one doesn't cause the problem

**Comment by SerhiiNahornyi** [comment]
@maxime-dupuis Could you pls merge master to this PR, it should fix tests

**Comment by maxime-dupuis** [comment]
> @maxime-dupuis Could you pls merge master to this PR, it should fix tests

@SerhiiNahornyi Thanks I merged it. I don't seem to be able to rerun the tests though, can you do it?

File: src/test/java/org/prebid/server/bidder/sharethrough/SharethroughBidderTest.java [hunk 1]
```diff
@@ -241,6 +244,94 @@ public void makeHttpRequestsShouldReturnErrorMessageOnFailedCurrencyConversion()
...
+                            assertThat(impression.getBanner()).isNull();
+                            assertThat(impression.getVideo()).isNull();
+                        });
+
+        // Ignore audio impressions because it is currently not supported
+        assertThat(httpRequests)
+                .extracting(HttpRequest::getPayload)
+                .extracting(BidRequest::getImp)
+                .extracting(impressions -> impressions.get(0))
+                .allSatisfy(impression -> assertThat(impression.getAudio()).isNull());
```
Comment (CTMBNara) [inline src/test/java/org/prebid/server/bidder/sharethrough/SharethroughBidderTest.java:310]: Please, assert audio object in previous blocks. For example:
```
                         impression -> {
                             assertThat(impression.getBanner()).isEqualTo(banner);
                             assertThat(impression.getVideo()).isNull();
                             assertThat(impression.getXNative()).isNull();
                             assertThat(impression.getAudio()).isNull();
                         },
```

**Review by CTMBNara — CHANGES_REQUESTED** [review]

**Review by CTMBNara — APPROVED** [review]

**Review by SerhiiNahornyi — APPROVED** [review]
LGTM

**Pull request merged**